crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -316,6 +335,21 @@ pub struct SyncTimelineEvent {
     pub push_actions: Vec<Action>,
 }
 
+const _: fn() = || {
+    fn assert_sync_and_send<T: ?Sized + Sync + Send>() {}
+
+    // `SyncTimelineEvent::kind` is of type `TimelineEventKind`.
+    assert_sync_and_send::<TimelineEventKind>();
+
+    // `SyncTimelineEvent::push_actions` is of type `Vec<Action>`.
+    assert_sync_and_send::<Vec<Action>>();
+};
+
+// All fields of `SyncTimelineEvent` implements `Sync` and `Send`: we can safely
+// implement `Sync` and `Send` for `SyncTimelineEvent` directly.
+unsafe impl Sync for SyncTimelineEvent {}
+unsafe impl Send for SyncTimelineEvent {}
```

Comment:
I think I like better the idea using the `test-send-sync` Cargo feature, that's disabled by default, but gets enabled in one configuration in CI; it avoids having to remember that if you add a new field to `SyncTimelineEvent`, then you need to update the above anonymous function doing the static assertion too.

https://github.com/matrix-org/matrix-rust-sdk/blob/ff7077b742d845dfd35f0730e7c2c75adf06b961/crates/matrix-sdk-base/src/rooms/normal.rs#L1264-L1279

CI test:

https://github.com/matrix-org/matrix-rust-sdk/blob/ff7077b742d845dfd35f0730e7c2c75adf06b961/xtask/src/ci.rs#L307

Reply:
Alright. I'm okay with both approaches.

`matrix-sdk-common` is a dependency of `matrix-sdk-base`. I've created the same feature flag on `matrix-sdk-common`, and `matrix-sdk-base` forwards its feature flag. All good!